benchmarks/benchmarks/stats.py

```diff
@@ -56,7 +56,99 @@ def time_ttest_ind_diff_var(self):
+    dists = sorted(list(set([d[0] for d in distcont + distdiscrete])))
+
+    param_names = ['distribution', 'properties']
+    params = [
+        dists, ['pdf/pmf', 'logpdf/logpmf', 'cdf', 'logcdf', 'rvs', 'fit',
+                'sf', 'logsf', 'ppf', 'isf', 'moment', 'stats_s', 'stats_v',
+                'stats_m', 'stats_k', 'stats_mvsk', 'entropy']
+    ]
+    dist_data = dict(distcont + distdiscrete)
+    # custom shape values can be provided for any distribution in the format
+    # `dist_name`: [shape1, shape2, ...]
+    custom_input = {}
+
+    # these are the distributions that are the slowest
+    slow_dists = ['nct', 'ncx2', 'argus', 'cosine', 'foldnorm', 'gausshyper',
+                  'kappa4', 'invgauss', 'wald', 'vonmises_line', 'ksone',
+                  'genexpon', 'exponnorm', 'recipinvgauss', 'vonmises',
+                  'foldcauchy', 'kstwo', 'levy_stable', 'skewnorm']
+    slow_methods = ['moment']
+
+    def setup(self, distribution, properties):
+        if not is_xslow() and (distribution in self.slow_dists or
+                               properties in self.slow_methods):
+            raise NotImplementedError("Skipped")
+
+        self.dist = getattr(stats, distribution)
+
+        shapes = self.dist_data[distribution]
+
+        if isinstance(self.dist, stats.rv_discrete):
+            # discrete distributions only use location
+            self.isCont = False
+            kwds = {'loc': 4}
+        else:
+            # continuous distributions use location and scale
+            self.isCont = True
+            kwds = {'loc': 4, 'scale': 10}
+
+        bounds = self.dist.interval(.99, *shapes, **kwds)
+        x = np.linspace(*bounds, 100)
+        args = [x, *self.custom_input.get(distribution, shapes)]
+
+        if properties == 'fit':
+            # the only positional argument is the data to be fitted
+            self.args = self.dist.rvs(size=100, **kwds)
+        elif properties == 'rvs':
+            # add size keyword argument for data creation
+            kwds['size'] = 1000
+            # keep shapes as positional arguments, omit linearly spaced data
+            self.args = args[1:]
+        elif properties == 'pdf/pmf':
+            properties = ('pmf' if isinstance(self.dist, stats.rv_discrete)
+                          else 'pdf')
+            self.args = args
+        elif properties == 'logpdf/logpmf':
+            properties = ('logpmf' if isinstance(self.dist, stats.rv_discrete)
+                          else 'logpdf')
+            self.args = args
+        elif properties in ['ppf', 'isf']:
+            self.args = [np.linspace((0, 1), 100), *args[1:]]
+        elif properties == 'moment':
+            # the first four moments may be optimized, so compute the fifth
+            self.args = [5, *args[1:]]
+        elif properties.startswith('stats_'):
+            properties = 'stats'
+            self.args = args[1:]
+            kwds['moments'] = properties[6:]
```

Comment:
`properties = 'stats'`, though? Do you mean to swap the order?

Reply:
I don't think I understand - the purpose of doing this was that the 'property', which come to think of it should probably be renamed to "method" for stats is passed in as 
 - 'stats_s',
 -  'stats_v',
 - 'stats_m',
 -  'stats_k',
 -  'stats_mvsk'

so it must be changed when getattr is used

Reply:
Ignore the middle line, then this says:
```
            properties = 'stats'
            kwds['moments'] = properties[6:]
```
That is not making sense to me. `properties[6:]` would be an empty string.

Also, I agree that another name would be better.

Reply:
I see what you meant by swap the order. Thanks